lib/stacks/iam.ts

```diff
@@ -396,19 +405,79 @@ export class IAMStack extends Stack {
             return statements;
         };
 
+        /*
+         * WARNING: Changing this method will cause any policy statement created by this to be regenerated. This will cause
+         * any changes to this policy (like dynamic policy updates for app configuration changes) to be lost until the app
+         * configuration is updated and it updates this policy with the expected values.
+         */
+        const instanceConstraintPolicyStatement = (partition: string, region: string, actionResourcePair: {[key: string]: string}) => {
+            const [actions, resources] = Object.entries(actionResourcePair).reduce(([actionAccumulator, resourcesAccumulator], [action, resource]) => {
+                return [[...actionAccumulator, `sagemaker:${action}`], [...resourcesAccumulator, `arn:${partition}:sagemaker:${region}:${this.account}:${resource}/*`]];
+            }, [[] as string[], [] as string[]]);
+
+            return [
+                new PolicyStatement({
+                    effect: Effect.ALLOW,
+                    actions,
+                    resources,
+                    conditions: {
+                        'ForAnyValue:StringEquals': {
+                            'sagemaker:InstanceTypes': [],
+                        },
+                    },
+                }),
+            ];
+        };
+
         const notebookPolicy = new ManagedPolicy(this, 'mlspace-notebook-policy', {
             statements: notebookPolicyStatements(this.partition, Aws.REGION),
             description: 'Enables general MLSpace actions in notebooks and across the entire application.'
         });
 
-        // If roles are manually created use the existing role
```

Comment:
Why remove this comment?

Reply:
Nope, no idea how that got removed. I'll add it back.